File: processingpy/CameraDrive/Controls.py
def setupControls(_cam):
    global cam, delta, keyW, keyA, keyS, keyD, keyQ, keyE, keySpace
    cam = _cam
    delta = 10
    keyW = False
    keyA = False
    keyS = False
    keyD = False
    keyQ = False
    keyE = False
    keySpace = False

def keyPressed():
    checkKeyChar(key, True)

def keyReleased():
    checkKeyChar(key, False)

def checkKeyChar(k, b):
    global keyW, keyA, keyS, keyD, keyQ, keyE, keySpace
    if (k == 'w'):
        keyW = b 
    elif (k == 'a'):
        keyA = b
    elif (k == 's'):
        keyS = b            
    elif (k == 'd'):
        keyD = b
    elif (k == 'q'):
        keyQ = b
    elif (k == 'e'):
        keyE = b
    elif (k == ' '):
        keySpace = b

def updateControls():
    global cam, keyW, keyA, keyS, keyD, keyQ, keyE, keySpace
    if (keyW):
        cam.move(0,0,-delta)
    if (keyS):
        cam.move(0,0,delta)
    if (keyA):
        cam.move(-delta,0,0)
    if (keyD):
        cam.move(delta,0,0)
    if (keyQ):
        cam.move(0,delta,0)
    if (keyE):
        cam.move(0,-delta,0)
    #~    
    if (keySpace):
        cam.reset()
        #keysOff() 

def keysOff():
    keyW = False
    keyA = False
    keyS = False
    keyD = False
    keyQ = False
    keyE = False
    keySpace = False

File: p5js/CameraDrive/js/controls.js
"use strict";

var deltaOrig = 10;
//var deltaFast = 10;
var delta = deltaOrig;

var keyW = false;
var keyA = false;
var keyS = false;
var keyD = false;
var keyQ = false;
var keyE = false;
var keySpace = false;

function keyPressed() {
  checkKeyChar(key, true);
}

function keyReleased() {
  checkKeyChar(key, false);
}

function checkKeyChar(k, b) {
  switch (k) {
    case 'w':
      return keyW = b;
    case 'a':
      return keyA = b;
    case 's':
      return keyS = b;      
    case 'd':
      return keyD = b;
    case 'q':
      return keyQ = b;
    case 'e':
      return keyE = b;
    case ' ':
      return keySpace = b;
    default:
      return b;
  }
}

/*
function checkKeyCode(int k, var b) {
  switch (k) {
    case SHIFT:
      return keyShift = b;
    default:
      return b;
  }
}
*/

function updateControls() {
  if (keyW) cam.move(0,0,-delta);
  if (keyS) cam.move(0,0,delta);
  if (keyA) cam.move(-delta,0,0);
  if (keyD) cam.move(delta,0,0);
  if (keyQ) cam.move(0,delta,0);
  if (keyE) cam.move(0,-delta,0);
  
  if (keySpace) {
    cam.reset();
    //keysOff();  
  }
}

function keysOff() {
  keyW = false;
  keyA = false;
  keyS = false;
  keyD = false;
  keyQ = false;
  keyE = false;
  keySpace = false;
}